File: src/Entity/Contenu.php
<?php
namespace App\Entity;

use App\Repository\ContenuRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ContenuRepository::class)]
class Contenu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    // type = "video", "pdf", "image", "description", "youtube"
    #[ORM\Column(type: 'string', length: 50)]
    private ?string $type = null;

    // Chemin du fichier si c'est video, pdf, image
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $fichier = null;

    // Texte si c'est "description"
    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    // ID de la vidéo YouTube si type = "youtube"
    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $youtubeId = null;

    #[ORM\ManyToOne(targetEntity: Niveau::class, inversedBy: 'contenus')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Niveau $niveau = null;

    public function getId(): ?int
    {
        return $this->id;
    }
    
    public function getType(): ?string
    {
        return $this->type;
    }
    
    public function setType(string $type): self
    {
        $this->type = $type;
        return $this;
    }
    
    public function getFichier(): ?string
    {
        return $this->fichier;
    }
    
    public function setFichier(?string $fichier): self
    {
        $this->fichier = $fichier;
        return $this;
    }
    
    public function getDescription(): ?string
    {
        return $this->description;
    }
    
    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }
    
    public function getYoutubeId(): ?string
    {
        return $this->youtubeId;
    }
    
    public function setYoutubeId(?string $youtubeId): self
    {
        $this->youtubeId = $youtubeId;
        return $this;
    }
    
    public function getNiveau(): ?Niveau
    {
        return $this->niveau;
    }
    
    public function setNiveau(?Niveau $niveau): self
    {
        $this->niveau = $niveau;
        return $this;
    }
}

File: src/main/java/entities/Contenu.java
package entities;

public class Contenu {
    private int id;
    private int niveauId;
    private String type;        // (vidéo upload, vidéo YouTube, PDF, image, description)
    private String fichier;     // fichier local ou PDF
    private String description; // texte libre
    private String youtubeId;   // si c’est une vidéo YouTube

    public Contenu() {
    }

    public Contenu(int niveauId, String type, String fichier, String description, String youtubeId) {
        this.niveauId = niveauId;
        this.type = type;
        this.fichier = fichier;
        this.description = description;
        this.youtubeId = youtubeId;
    }

    public Contenu(int id, int niveauId, String type, String fichier, String description, String youtubeId) {
        this.id = id;
        this.niveauId = niveauId;
        this.type = type;
        this.fichier = fichier;
        this.description = description;
        this.youtubeId = youtubeId;
    }

    public int getId() {
        return id;
    }

    public int getNiveauId() {
        return niveauId;
    }

    public void setNiveauId(int niveauId) {
        this.niveauId = niveauId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getFichier() {
        return fichier;
    }

    public void setFichier(String fichier) {
        this.fichier = fichier;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getYoutubeId() {
        return youtubeId;
    }

    public void setYoutubeId(String youtubeId) {
        this.youtubeId = youtubeId;
    }
}
